src/main/java/com/example/gsoc/controller/PluginController.java

```diff
@@ -0,0 +1,69 @@
+package com.example.gsoc.controller;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.example.gsoc.model.Plugin;
+import com.example.gsoc.model.UpdateCenter;
+import com.example.gsoc.service.PluginService;
+import com.fasterxml.jackson.core.exc.StreamReadException;
+import com.fasterxml.jackson.databind.DatabindException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+public class PluginController {
+	@Autowired
+	PluginService pluginService;
+	
+	public PluginController() throws StreamReadException, DatabindException, IOException {
+		readUpdateCenter();
+	}
+	
+	public List<Plugin> readUpdateCenter() throws StreamReadException, DatabindException, IOException {
```

Comment:
@alecharp should I move this method to PluginService.java? Since I read business logic should be inside the Service component.

Reply:
Good point. Go for it.